### PR TITLE
dev-debug/sysdig:revbump for libs-0.17.3, do not install dev headers/libs

### PR DESCRIPTION
**Pull request by hhoffstaette**
A minor update to falcosecurity-libs-0.17.3, fixing a few issues since 0.17.2.
Building sysdig with LTO now works too, with both gcc and clang.
No longer install 35 MB of headers & libs which are not very usable, see [here](https://bugs.gentoo.org/938187) for why.

I'd like to stable this ASAP and then get rid of all older versions.

---

Please check all the boxes that apply:

- [x] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [x] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [x] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [x] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

### CODE REVIEW

File: dev-debug/sysdig/sysdig-0.38.1-r1.ebuild [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+src_install() {
+	cmake_src_install
+
+	# remove driver headers
+	rm -r "${ED}"/usr/src || die
+
+	# remove libscap/libsinsp headers & libs unless requested
+	if ! use libs; then
+		rm -r "${ED}"/usr/include/sysdig || die
+		rm -r "${ED}"/usr/lib64 || die
```
Comment (floppym) [inline dev-debug/sysdig/sysdig-0.38.1-r1.ebuild:128]: This is not going to work on x86, where libdir != "lib64".

**Review by floppym — CHANGES_REQUESTED** [review]
Making libs/headers optional is not something we normally do. For a debugging tool, the disk space savings seems a bit pointless.

**Comment by eli-schwartz** [comment]
> Making libs/headers optional is not something we normally do. For a debugging tool, the disk space savings seems a bit pointless.

The actual libraries themselves appear to be compiled and installed by falcosecurity/libs, rather than sysdig itself. I wonder what their overall purpose is as a subcomponent of sysdig. Maybe the real answer is it should be a DEPEND?

When does one want these libraries, and when not?

File: dev-debug/sysdig/sysdig-0.38.1-r1.ebuild (hunk 1 above)
Comment (hhoffstaette) [reply dev-debug/sysdig/sysdig-0.38.1-r1.ebuild:128]: Gah, fair point. I'll rework this to use get_libdir.

Comment (hhoffstaette) [reply dev-debug/sysdig/sysdig-0.38.1-r1.ebuild]: Fixed, thanks.

**Comment by hhoffstaette** [comment]
One immediate example for why these libs are not really usable in their current form is e.g. https://github.com/falcosecurity/libs/issues/2004

**Comment by eli-schwartz** [comment]
> One immediate example for why these libs are not really usable in their current form is e.g. [falcosecurity/libs#2004](https://github.com/falcosecurity/libs/issues/2004)

Genuinely horrifying. It should be illegal to use CMake as a build system.

**Comment by hhoffstaette** [comment]
USE=libs removed. Should have done that from the beginning..

**Review by floppym — APPROVED** [review]

**Comment by eli-schwartz** [comment]
```
$ readelf -d /usr/bin/sysdig
 0x0000000000000001 (NEEDED)             Shared library: [libabsl_synchronization.so.2206.0.0]
```

Seems that's being exposed by grpc? iwdevtools complains:
```
 * VDB: detected possibly incorrect RDEPEND (dev-debug/sysdig-0.38.1-r1)
 *                  > dev-cpp/abseil-cpp
 * dev-libs/openssl < 
 * net-dns/c-ares   < 
```

(dunno about the left hand side ones. Those can be prone to false positives especially if something uses the CLI tools instead -- but to be fair, sysdig is binding to subslot here so perhaps it is indeed incorrect.)

**Comment by hhoffstaette** [comment]
> ```
> $ readelf -d /usr/bin/sysdig
>  0x0000000000000001 (NEEDED)             Shared library: [libabsl_synchronization.so.2206.0.0]
> ```

The dependencies are a huge mess and have always been hit & miss, thanks to the libs project and partial vendoring that varies from release to release. Don't try to understand them.
In this case it's transitively from both grpc and re2:
```
$lddtree /usr/bin/sysdig 
sysdig => /usr/bin/sysdig (interpreter => /lib64/ld-linux-x86-64.so.2)
    libluajit-5.1.so.2 => /usr/lib64/libluajit-5.1.so.2
        ld-linux-x86-64.so.2 => /lib64/ld-linux-x86-64.so.2
    libyaml-cpp.so.0.8 => /usr/lib64/libyaml-cpp.so.0.8
    libelf.so.1 => /usr/lib64/libelf.so.1
        libzstd.so.1 => /usr/lib64/libzstd.so.1
    libz.so.1 => /usr/lib64/libz.so.1
    libcurl.so.4 => /usr/lib64/libcurl.so.4
        libcares.so.2 => /usr/lib64/libcares.so.2
        libnghttp3.so.9 => /usr/lib64/libnghttp3.so.9
        libnghttp2.so.14 => /usr/lib64/libnghttp2.so.14
        libpsl.so.5 => /usr/lib64/libpsl.so.5
            libidn2.so.0 => /usr/lib64/libidn2.so.0
            libunistring.so.5 => /usr/lib64/libunistring.so.5
        libssl.so.3 => /usr/lib64/libssl.so.3
        libcrypto.so.3 => /usr/lib64/libcrypto.so.3
    libjsoncpp.so.25 => /usr/lib64/libjsoncpp.so.25
    libre2.so.11 => /usr/lib64/libre2.so.11
        libabsl_log_internal_conditions.so.2401.0.0 => /usr/lib64/libabsl_log_internal_conditions.so.2401.0.0
        libabsl_examine_stack.so.2401.0.0 => /usr/lib64/libabsl_examine_stack.so.2401.0.0
        libabsl_log_internal_format.so.2401.0.0 => /usr/lib64/libabsl_log_internal_format.so.2401.0.0
        libabsl_log_internal_proto.so.2401.0.0 => /usr/lib64/libabsl_log_internal_proto.so.2401.0.0
        libabsl_log_internal_log_sink_set.so.2401.0.0 => /usr/lib64/libabsl_log_internal_log_sink_set.so.2401.0.0
        libabsl_log_internal_globals.so.2401.0.0 => /usr/lib64/libabsl_log_internal_globals.so.2401.0.0
        libabsl_log_globals.so.2401.0.0 => /usr/lib64/libabsl_log_globals.so.2401.0.0
        libabsl_log_sink.so.2401.0.0 => /usr/lib64/libabsl_log_sink.so.2401.0.0
        libabsl_log_entry.so.2401.0.0 => /usr/lib64/libabsl_log_entry.so.2401.0.0
        libabsl_strerror.so.2401.0.0 => /usr/lib64/libabsl_strerror.so.2401.0.0
        libabsl_vlog_config_internal.so.2401.0.0 => /usr/lib64/libabsl_vlog_config_internal.so.2401.0.0
        libabsl_log_internal_fnmatch.so.2401.0.0 => /usr/lib64/libabsl_log_internal_fnmatch.so.2401.0.0
        libabsl_flags_internal.so.2401.0.0 => /usr/lib64/libabsl_flags_internal.so.2401.0.0
        libabsl_flags_marshalling.so.2401.0.0 => /usr/lib64/libabsl_flags_marshalling.so.2401.0.0
        libabsl_flags_reflection.so.2401.0.0 => /usr/lib64/libabsl_flags_reflection.so.2401.0.0
        libabsl_flags_private_handle_accessor.so.2401.0.0 => /usr/lib64/libabsl_flags_private_handle_accessor.so.2401.0.0
        libabsl_flags_commandlineflag.so.2401.0.0 => /usr/lib64/libabsl_flags_commandlineflag.so.2401.0.0
        libabsl_flags_commandlineflag_internal.so.2401.0.0 => /usr/lib64/libabsl_flags_commandlineflag_internal.so.2401.0.0
        libabsl_flags_config.so.2401.0.0 => /usr/lib64/libabsl_flags_config.so.2401.0.0
        libabsl_flags_program_name.so.2401.0.0 => /usr/lib64/libabsl_flags_program_name.so.2401.0.0
        libabsl_cord_internal.so.2401.0.0 => /usr/lib64/libabsl_cord_internal.so.2401.0.0
        libabsl_cordz_handle.so.2401.0.0 => /usr/lib64/libabsl_cordz_handle.so.2401.0.0
        libabsl_crc_cord_state.so.2401.0.0 => /usr/lib64/libabsl_crc_cord_state.so.2401.0.0
        libabsl_crc32c.so.2401.0.0 => /usr/lib64/libabsl_crc32c.so.2401.0.0
        libabsl_crc_internal.so.2401.0.0 => /usr/lib64/libabsl_crc_internal.so.2401.0.0
        libabsl_crc_cpu_detect.so.2401.0.0 => /usr/lib64/libabsl_crc_cpu_detect.so.2401.0.0
        libabsl_raw_hash_set.so.2401.0.0 => /usr/lib64/libabsl_raw_hash_set.so.2401.0.0
        libabsl_hashtablez_sampler.so.2401.0.0 => /usr/lib64/libabsl_hashtablez_sampler.so.2401.0.0
        libabsl_exponential_biased.so.2401.0.0 => /usr/lib64/libabsl_exponential_biased.so.2401.0.0
        libabsl_city.so.2401.0.0 => /usr/lib64/libabsl_city.so.2401.0.0
        libabsl_bad_variant_access.so.2401.0.0 => /usr/lib64/libabsl_bad_variant_access.so.2401.0.0
        libabsl_low_level_hash.so.2401.0.0 => /usr/lib64/libabsl_low_level_hash.so.2401.0.0
        libabsl_bad_optional_access.so.2401.0.0 => /usr/lib64/libabsl_bad_optional_access.so.2401.0.0
        libabsl_str_format_internal.so.2401.0.0 => /usr/lib64/libabsl_str_format_internal.so.2401.0.0
        libabsl_graphcycles_internal.so.2401.0.0 => /usr/lib64/libabsl_graphcycles_internal.so.2401.0.0
        libabsl_kernel_timeout_internal.so.2401.0.0 => /usr/lib64/libabsl_kernel_timeout_internal.so.2401.0.0
        libabsl_stacktrace.so.2401.0.0 => /usr/lib64/libabsl_stacktrace.so.2401.0.0
        libabsl_symbolize.so.2401.0.0 => /usr/lib64/libabsl_symbolize.so.2401.0.0
        libabsl_debugging_internal.so.2401.0.0 => /usr/lib64/libabsl_debugging_internal.so.2401.0.0
        libabsl_demangle_internal.so.2401.0.0 => /usr/lib64/libabsl_demangle_internal.so.2401.0.0
        libabsl_malloc_internal.so.2401.0.0 => /usr/lib64/libabsl_malloc_internal.so.2401.0.0
        libabsl_time.so.2401.0.0 => /usr/lib64/libabsl_time.so.2401.0.0
        libabsl_civil_time.so.2401.0.0 => /usr/lib64/libabsl_civil_time.so.2401.0.0
        libabsl_strings.so.2401.0.0 => /usr/lib64/libabsl_strings.so.2401.0.0
        libabsl_strings_internal.so.2401.0.0 => /usr/lib64/libabsl_strings_internal.so.2401.0.0
        libabsl_string_view.so.2401.0.0 => /usr/lib64/libabsl_string_view.so.2401.0.0
        libabsl_base.so.2401.0.0 => /usr/lib64/libabsl_base.so.2401.0.0
        libabsl_spinlock_wait.so.2401.0.0 => /usr/lib64/libabsl_spinlock_wait.so.2401.0.0
        libabsl_int128.so.2401.0.0 => /usr/lib64/libabsl_int128.so.2401.0.0
        libabsl_throw_delegate.so.2401.0.0 => /usr/lib64/libabsl_throw_delegate.so.2401.0.0
        libabsl_raw_logging_internal.so.2401.0.0 => /usr/lib64/libabsl_raw_logging_internal.so.2401.0.0
        libabsl_log_severity.so.2401.0.0 => /usr/lib64/libabsl_log_severity.so.2401.0.0
        libabsl_time_zone.so.2401.0.0 => /usr/lib64/libabsl_time_zone.so.2401.0.0
        libicuuc.so.75 => /usr/lib64/libicuuc.so.75
        libicudata.so.75 => /usr/lib64/libicudata.so.75
    libgrpc++.so.1.65 => /usr/lib64/libgrpc++.so.1.65
        libupb_json_lib.so.42 => /usr/lib64/libupb_json_lib.so.42
            libutf8_range_lib.so.42 => /usr/lib64/libutf8_range_lib.so.42
            libupb_message_lib.so.42 => /usr/lib64/libupb_message_lib.so.42
            libupb_base_lib.so.42 => /usr/lib64/libupb_base_lib.so.42
        libupb_mem_lib.so.42 => /usr/lib64/libupb_mem_lib.so.42
        libabsl_statusor.so.2401.0.0 => /usr/lib64/libabsl_statusor.so.2401.0.0
        libabsl_status.so.2401.0.0 => /usr/lib64/libabsl_status.so.2401.0.0
    libprotobuf.so.27.2.0 => /usr/lib64/libprotobuf.so.27.2.0
        libabsl_die_if_null.so.2401.0.0 => /usr/lib64/libabsl_die_if_null.so.2401.0.0
    libgrpc.so.42 => /usr/lib64/libgrpc.so.42
        libupb_textformat_lib.so.42 => /usr/lib64/libupb_textformat_lib.so.42
        libaddress_sorting.so.42 => /usr/lib64/libaddress_sorting.so.42
        libabsl_random_internal_pool_urbg.so.2401.0.0 => /usr/lib64/libabsl_random_internal_pool_urbg.so.2401.0.0
            libabsl_random_internal_seed_material.so.2401.0.0 => /usr/lib64/libabsl_random_internal_seed_material.so.2401.0.0
            libabsl_random_seed_gen_exception.so.2401.0.0 => /usr/lib64/libabsl_random_seed_gen_exception.so.2401.0.0
        libabsl_random_internal_randen.so.2401.0.0 => /usr/lib64/libabsl_random_internal_randen.so.2401.0.0
            libabsl_random_internal_randen_hwaes.so.2401.0.0 => /usr/lib64/libabsl_random_internal_randen_hwaes.so.2401.0.0
        libabsl_random_internal_randen_hwaes_impl.so.2401.0.0 => /usr/lib64/libabsl_random_internal_randen_hwaes_impl.so.2401.0.0
            libabsl_random_internal_platform.so.2401.0.0 => /usr/lib64/libabsl_random_internal_platform.so.2401.0.0
        libabsl_random_internal_randen_slow.so.2401.0.0 => /usr/lib64/libabsl_random_internal_randen_slow.so.2401.0.0
    libabsl_log_internal_check_op.so.2401.0.0 => /usr/lib64/libabsl_log_internal_check_op.so.2401.0.0
    libabsl_log_internal_message.so.2401.0.0 => /usr/lib64/libabsl_log_internal_message.so.2401.0.0
    libabsl_log_internal_nullguard.so.2401.0.0 => /usr/lib64/libabsl_log_internal_nullguard.so.2401.0.0
    libabsl_cord.so.2401.0.0 => /usr/lib64/libabsl_cord.so.2401.0.0
    libabsl_cordz_info.so.2401.0.0 => /usr/lib64/libabsl_cordz_info.so.2401.0.0
    libabsl_cordz_functions.so.2401.0.0 => /usr/lib64/libabsl_cordz_functions.so.2401.0.0
    libabsl_hash.so.2401.0.0 => /usr/lib64/libabsl_hash.so.2401.0.0
    libabsl_synchronization.so.2401.0.0 => /usr/lib64/libabsl_synchronization.so.2401.0.0
    libgpr.so.42 => /usr/lib64/libgpr.so.42
    libm.so.6 => /lib64/libm.so.6
    libtbb.so.12 => /usr/lib64/libtbb.so.12
    libstdc++.so.6 => /usr/lib/gcc/x86_64-pc-linux-gnu/14/libstdc++.so.6
    libgcc_s.so.1 => /usr/lib/gcc/x86_64-pc-linux-gnu/14/libgcc_s.so.1
    libc.so.6 => /lib64/libc.so.6
```
There does not seem to be a direct dependency (via include) of any absl headers, however IIRC both grpc and re2 bleed absl types out of their headers (despite their own claims of not doing so).

For good measure the final link step e.g. for the sysdig executable also adds libraries both as -llibname *and* as direct object, maybe "to make sure" - or nobody ever noticed. Again, don't try to understand any of this.

> Seems that's being exposed by grpc? iwdevtools complains:
> 
> ```
>  * VDB: detected possibly incorrect RDEPEND (dev-debug/sysdig-0.38.1-r1)
>  *                  > dev-cpp/abseil-cpp
>  * dev-libs/openssl < 
>  * net-dns/c-ares   < 
> ```

If it makes QA happy I can add absl as RDEPEND. openssl/c-ares are transitive dependencies of libcurl and can probably be removed. I'll try that tomorrow.

**Comment by eli-schwartz** [comment]
> In this case it's transitively from both grpc and re2:

That would not be an issue to the QA check, which only cares about lddtree results with a single indent, not lddtree results with multiple indentation levels.

> There does not seem to be a direct dependency (via include) of any absl headers, however IIRC both grpc and re2 bleed absl types out of their headers (despite their own claims of not doing so).
> 
> For good measure the final link step e.g. for the sysdig executable also adds libraries both as -llibname _and_ as direct object, maybe "to make sure" - or nobody ever noticed. Again, don't try to understand any of this.

Yup, this right here is the reason why it's a direct dependency of the sysdig binary itself. I won't try to understand why it adds them both as -llibname and as /path/to/libfoo.so but given that they ultimately have abseil types in their compiled objects they do need to link to it directly somehow -- and that's exactly what they do.

> If it makes QA happy I can add absl as RDEPEND.

The QA script is really just warning you that in the event abseil-cpp gets updated, grpc and re2 will undergo a subslot rebuild but sysdig will not, which means that sysdig will fail to start with a missing library (or if preserved-libs is in effect, it may load multiple incompatible versions of libabsl_* libraries and crash instead).

**Comment by hhoffstaette** [comment]
> The QA script is really just warning you that in the event abseil-cpp gets updated, grpc and re2 will undergo a subslot rebuild but sysdig will not, (..)

:bulb:
So if I understand correctly it should be enough for the absl/slot dependency to be `dev-cpp/abseil-cpp:=`, correct? Or do I really need to depend on the exact same version as the grpc/protobuf/re2 dependencies do themselves?

**Comment by eli-schwartz** [comment]
The former.

Both sysdig and grpc will bind to "whichever version is installed at the time of building". Changing the abseil version for grpc means grpc will be rebuilt against whichever version is installed, so grpc and abseil will always be in sync.

And sysdig will therefore be in sync with whichever version of abseil was installed at the same time that "grpc is also installed" plus "sysdig got built" -- the subslot just forces you to rebuild sysdig every time the abseil-cpp package changes.

**Pull request closed without merging**